tests/ProducerTest.cc

```diff
@@ -373,4 +374,62 @@ TEST_P(ProducerTest, testFlushNoBatch) {
     client.close();
 }
 
+TEST_P(ProducerTest, testCloseSubProducerWhenFail) {
```

Comment:
```suggestion
TEST(ProducerTest, testCloseSubProducerWhenFail) {
```

No need to use `TEST_P` here.